http/cookie.ts

```diff
@@ -218,3 +218,97 @@ export function deleteCookie(
     ...attributes,
   });
 }
+
+/**
+ * @param value
+ * @returns
+ */
+function parseSetCookie(value: string): Cookie {
+  const attrs = value
+    .split(";")
+    .map((attr) =>
+      attr
+        .trim()
+        .split("=")
+        .map((keyOrValue) => keyOrValue.trim())
+    );
+  const cookie: Cookie = {
+    name: attrs[0][0],
+    value: attrs[0][1],
+  };
+
+  for (const [key, value] of attrs.slice(1)) {
+    switch (key.toLocaleLowerCase()) {
+      case "expires":
+        cookie.expires = new Date(value);
+        break;
+      case "max-age":
+        cookie.maxAge = Number(value);
+        assert(
+          cookie.maxAge >= 0,
+          "Max-Age must be an integer superior or equal to 0",
+        );
+        break;
+      case "domain":
+        cookie.domain = value;
+        break;
+      case "path":
+        cookie.path = value;
+        break;
+      case "secure":
+        cookie.secure = true;
+        break;
+      case "httponly":
+        cookie.httpOnly = true;
+        break;
+      case "samesite":
+        cookie.sameSite = value as Cookie["sameSite"];
+        break;
+      default:
+        if (!Array.isArray(cookie.unparsed)) {
+          cookie.unparsed = [];
+        }
+        cookie.unparsed.push([key, value].join("="));
+    }
+  }
+  if (cookie.name.startsWith("__Secure-")) {
+    /** This requirement is mentioned in https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie but not the RFC. */
+    assert(
+      cookie.secure,
+      "Cookies with names starting with `__Secure-` must be set with the secure flag.",
+    );
```

Comment:
I feel it's more user-friendly to show this message as warning and skip this 'set-cookie' value. I believe browsers work like that. (The same applies to the below 3 asserts)

What do you thinkg?

Reply:
Yeah, good point. Perhaps `console.assert` is more fitting. What do you think?

Reply:
Agree. `console.assert` works as well.

Reply:
It turns out using `console.assert` might not be workable. My latest change instead uses `console.warn`. I'm unsure whether it's the best way to do it, but at least invalid cookies aren't thrown now. And I'm not sure whether logging to the console is desirable in tests and such... What do you think?